app/src/main/java/com/humberto/tasky/main/NavigationRoot.kt

```diff
@@ -90,6 +92,8 @@ private fun NavGraphBuilder.agendaGraph(navController: NavHostController) {
     }
     composable<AgendaDetails> { backStackEntry ->
         val agendaDetails: AgendaDetails = backStackEntry.toRoute<AgendaDetails>()
+        val title = backStackEntry.savedStateHandle.get<String>("title")
+        val description = backStackEntry.savedStateHandle.get<String>("description")
```

Comment:
This is how you retirieve this later:

```
val key = editTextScreen.editTextScreenType.name.lowercase()
  navController.previousBackStackEntry
    ?.savedStateHandle
    ?.set(key, editTextScrzeen.content)
```

Could it also not make sense to use that enum to set the key values instead of using hardcoded Strings?

Reply:
You mean I also pass the "key" to the savedStateHandle? Or literally use editTextScreen.editTextScreenType enum on AgendaDetails navGraph? I tried a few things but wasn't able to get to a good solution.

Reply:
I pointed out that you use the enum EditTextScreenType (I think that is the correct name) as the key value. So I was wondering why you wouldn’t also use the enum to retrieve the data.

That way, you don’t have a hardcoded string, and the solution is a bit more scalable and less temporally coupled. In short - adding a new enum value already supports passing it as a key without needing that you fetch the data via a hardcoded string key.